python/ray/workers/setup_cgroup.py

```diff
@@ -0,0 +1,101 @@
+import errno
+import json
+import os
+import random
+import string
+
+cgroup_v1_root = "/sys/fs/cgroup"
+ray_parent_cgroup_name = "ray"
+
+
+# The parent cgroup can help to limit the resource all ray workers can use
+def create_ray_parent_cgroup(controller):
+    ray_parent_cgroup = os.path.join(cgroup_v1_root, controller, ray_parent_cgroup_name)
+    try:
+        os.mkdir(ray_parent_cgroup)
+        if controller == "cpuset":
+            cpuset_root = os.path.join(cgroup_v1_root, controller)
+            cpus = get_ray_cgroup_property(cpuset_root, "cpuset.cpus")
+            mems = get_ray_cgroup_property(cpuset_root, "cpuset.mems")
+            set_ray_cgroup_property(ray_parent_cgroup, "cpuset.cpus", cpus)
+            set_ray_cgroup_property(ray_parent_cgroup, "cpuset.mems", mems)
+        return ray_parent_cgroup
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
+
+
+def create_ray_cgroup(controller, cgroup_name):
+    ray_cgroup = os.path.join(cgroup_v1_root, controller, ray_parent_cgroup_name, cgroup_name)
+    os.mkdir(ray_cgroup)
+    return ray_cgroup
+
+
+def set_ray_cgroup_property(cgroup_path, property, data):
+    filename = os.path.join(cgroup_path, property)
+    with open(filename, "w") as f:
+        return f.write(str(data))
+
+
+def get_ray_cgroup_property(cgroup_path, property):
+    filename = os.path.join(cgroup_path, property)
+    with open(filename) as f:
+        return f.read().strip()
+
+
+def delete_cgroup_path(cgroup_path):
+    if os.path.exists(cgroup_path):
+        os.rmdir(cgroup_path)
+
+
+def create_cgroup_for_worker(resource_json):
+    # todo add timestampe?
+    cgroup_name = "ray-" + "".join(
+        random.sample(string.ascii_letters + string.digits, 8))
+    allocated_resource = json.loads(resource_json)
```

Comment:
Validate it?

Reply:
Maybe reusing some functions we use for ray.init

Reply:
Do you mean the `resource_json`?  It is generated by raylet, maybe we don't need to validate it?